test/acceptance/tests/mesh-gateway/mesh_gateway_test.go

```diff
@@ -0,0 +1,250 @@
+package meshgateway
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/hashicorp/consul-helm/test/acceptance/framework"
+	"github.com/hashicorp/consul-helm/test/acceptance/helpers"
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/sdk/testutil/retry"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// Test that Connect and wan federation over mesh gateways work in a default installation
+// i.e. without ACLs because TLS is required for WAN federation over mesh gateways
+func TestMeshGatewayDefault(t *testing.T) {
+	env := suite.Environment()
+
+	primaryHelmValues := map[string]string{
+		"global.datacenter":                        "dc1",
+		"global.tls.enabled":                       "true",
+		"global.tls.httpsOnly":                     "false",
+		"global.federation.enabled":                "true",
+		"global.federation.createFederationSecret": "true",
+
+		"connectInject.enabled": "true",
+
+		"meshGateway.enabled":  "true",
+		"meshGateway.replicas": "1",
+	}
+
+	releaseName := helpers.RandomName()
+
+	// Install the primary consul cluster in the default kubernetes context
+	primaryConsulCluster := framework.NewHelmCluster(t, primaryHelmValues, env.DefaultContext(t), suite.Config(), releaseName)
+	primaryConsulCluster.Create(t)
+
+	secondaryContext := env.Context(t, framework.SecondaryContextName)
+
+	// Get the federation secret from the primary cluster and apply it to secondary cluster
+	federationSecretName := fmt.Sprintf("%s-consul-federation", releaseName)
+	t.Logf("retrieving federation secret %s from the primary cluster and applying to the secondary", federationSecretName)
+	secretYAML, err := helpers.RunKubectlAndGetOutputWithLoggerE(t, env.DefaultContext(t).KubectlOptions(), logger.Discard, "get", "secret", federationSecretName, "-o", "yaml")
+	require.NoError(t, err)
+	helpers.KubectlApplyFromString(t, secondaryContext.KubectlOptions(), secretYAML)
```

Comment:
This is more of a personal preference but could we use the k8s client to perform this operation. Using a client to manage resources feels more robust than shell-ing out. Roughly something that looks like:
```
	federationSecret, err := env.DefaultContext(t).KubernetesClient(t).CoreV1().Secrets(env.DefaultContext(t).KubectlOptions().Namespace).Get(federationSecretName, metav1.GetOptions{})
	require.NoError(t, err)
	_, err = secondaryContext.KubernetesClient(t).CoreV1().Secrets(secondaryContext.KubectlOptions().Namespace).Create(federationSecret)
	require.NoError(t, err)
```

Reply:
Yeah! I think I just codified our docs and didn't think to convert the client here 🤦‍♀️ . Thanks!

Reply:
This is why we ~pair~ review code 😉

Reply:
Fixed! I had to set resourceVersion to `""` on the secret before applying it to another cluster, otherwise, kubernetes would reject the request, but I think it's ok and still better than shelling out.